code/game/machinery/tcomms/tcomms_base.dm

```diff
@@ -315,6 +315,9 @@ GLOBAL_LIST_EMPTY(tcomms_machines)
 					// Only add if it wasnt there already
 					radios |= R
 
+	// This is what makes dchat get their radio coverage. Do not remove.
+	radios += GLOB.deadchat_radio
+
```

Comment:
Can you initialize `radios` with this on line 283? `radios = list(GLOB.deadchat_radio)`